lib/iris/cube.py

```diff
@@ -758,6 +767,231 @@ def _is_single_item(testee):
     return isinstance(testee, str) or not isinstance(testee, Iterable)
 
 
+class CubeAttrsDict(MutableMapping):
```

Comment:
I think so, yes, because pretending that attributes are a simple dictionary only works 'up to a point', and for user-defined attributes we would intend future user code to create attributes of a specific 'type'.
I have also chosen to make it [visible in the object reprs](https://github.com/SciTools/iris/blob/7db42ce754e1d6d8b22f69aed2baa3eb4406c250/lib/iris/cube.py#L814-L815) : we now see `CubeAttrsDict`as a class name.  
Please say, if you think we should instead hide this ...